rpc/handlers.go

```diff
@@ -1126,3 +1128,22 @@ func (h *Handler) EstimateFee(broadcastedTxns []BroadcastedTransaction, id Block
 
 	return estimates, nil
 }
+
+func (h *Handler) EstimateMessageFee(msg MsgFromL1, id BlockID) (*FeeEstimate, *jsonrpc.Error) {
+	tx := BroadcastedTransaction{
+		Transaction: Transaction{
+			Type:               TxnL1Handler,
+			ContractAddress:    msg.To,
+			EntryPointSelector: msg.Selector,
+			CallData:           &msg.Payload,
+			Version: new(felt.Felt), // Needed for transaction hash calculation.
+			Nonce: new(felt.Felt), // Needed for transaction hash calculation.
+		},
+		PaidFeeOnL1: new(felt.Felt), // Needed to marshal to blockifier type.
+	}
+	estimates, rpcErr := h.EstimateFee([]BroadcastedTransaction{tx}, id)
```

Comment:
Wouldn't using `vm.Execute` directly instead be easier?

Reply:
Probably. Just hacked this together to get it working. Will refactor once I can reliably execute L1 handler transactions.

Reply:
On second thought, I think it makes sense to call `h.EstimateFee`. Almost all the other logic would be repeated otherwise.